melodies_monet/plots/surfplots.py

```diff
@@ -769,3 +906,145 @@ def make_boxplot(comb_bx, label_bx, ylabel = None, vmin = None, vmax = None, out
     
     plt.tight_layout()
     savefig(outname + '.png', loc=4, logo_height=100)
+
+def make_spatial_exceedance(df, column_o=None, label_o=None, column_m=None,
+                      label_m=None, ylabel = None, vdiff=None,
+                      outname = 'plot',
+                      domain_type=None, domain_name=None, fig_dict=None,
+                      text_dict=None,debug=False):
+
+    """Creates surface spatial bias plot. 
+    
+    Parameters
+    ----------
+    df : dataframe
+        model/obs pair data to plot
+    column_o : str
+        Column label of observation variable to plot
+    label_o : str
+        Name of observation variable to use in plot title 
+    column_m : str
+        Column label of model variable to plot
+    label_m : str
+        Name of model variable to use in plot title
+    ylabel : str
+        Title of colorbar axis
+    vdiff : real number
+        Min and max value to use on colorbar axis
+    outname : str
+        file location and name of plot (do not include .png)
+    domain_type : str
+        Domain type specified in input yaml file
+    domain_name : str
+        Domain name specified in input yaml file
+    fig_dict : dictionary
+        Dictionary containing information about figure
+    text_dict : dictionary
+        Dictionary containing information about text
+    debug : boolean
+        Whether to plot interactively (True) or not (False). Flag for 
+        submitting jobs to supercomputer turn off interactive mode.
+        
+    Returns
+    -------
+    plot 
+        surface bias plot
+    """
+    if debug == False:
+        plt.ioff()
+
+    def_map = dict(states=True,figsize=[10, 5])
+    if fig_dict is not None:
+        map_kwargs = {**def_map, **fig_dict}
+    else:
+        map_kwargs = def_map
+
+    #If not specified use the PlateCarree projection
+    if 'crs' not in map_kwargs:
+        map_kwargs['crs'] = ccrs.PlateCarree()
+
+    #set default text size
+    def_text = dict(fontsize=20)
+    if text_dict is not None:
+        text_kwargs = {**def_text, **text_dict}
+    else:
+        text_kwargs = def_text
+
+    # set ylabel to column if not specified.
+    if ylabel is None:
+        ylabel = column_o
+
+    #Take the mean for each siteid
+    #df_mean=df.groupby(['siteid'],as_index=False).mean()
+
+    #df_reg = pd.DataFrame()
+    print(df)
+    # calculate exceedance
+    if column_o == 'OZONE_reg':
+        # JianHe: we can get mean, or 95th, 99th percentile
```

Comment:
If you make the spatial_exceedance a plot_type instead of an add on to the spatial bias plot. You can add options in the yaml file calculating the mean or percentiles here if needed.

If this is a separate plotting group it also enables more flexibility for the axis. You can make it possible to use a different default max or even add one to the list like vmax_ct or something. And in the yaml file if you set set_axis: False then you can configure it so that it just uses a default vmax value, which may be the best for plots like this.